### PR TITLE
feat(bulk-import): preview catalog-info.yaml on a side-panel

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**

https://issues.redhat.com/browse/RHIDP-1666

**GIF:**

https://github.com/janus-idp/backstage-plugins/assets/22490998/00d1ec6a-2211-4ca9-a486-281820920320

**Test coverage:**
Screenshot 2024-06-21 at 1 59 22 PM
Screenshot 2024-06-21 at 1 59 14 PM

### CODE REVIEW

**Comment by ShiranHi** [comment]
@debsmita1 looks good to me, thank you!

**Review by ciiay — COMMENTED** [review]
It works as expected, just found some console error when update `Entity owner` in the side panel. If it's too complicated to fix we can address it in a different ticket.
image

**Review by ciiay — APPROVED** [review]
/lgtm

**Comment by invincibleJai** [comment]
Thanks @debsmita1, i have below observations

image


- Should the buttons on the sidebar footer be sticky? 
- Will sidebar content change depending on GIt or service now selection
- Seeing warnings on load

**Comment by debsmita1** [comment]
> Thanks @debsmita1, i have below observations
> 
> image
> * Should the buttons on the sidebar footer be sticky?
> * Will sidebar content change depending on GIt or service now selection
> * Seeing warnings on load

@ShiranHi Could you help answer the first 2 questions ?

**Comment by ShiranHi** [comment]
> * Should the buttons on the sidebar footer be sticky?
> * Will sidebar content change depending on GIt or service now selection

1. Yes, the content is long and we want to make the buttons clearly visible, [so I asked to make the buttons sticky for side panels](https://www.figma.com/design/09Iltd57RiuqOtKWkjy8MH/Bulk-import?node-id=74-11100&t=N2az2IcEinwIW3mN-1)
2. The content should be the same, I guess it will be helpful to change the wording "pull request" to be "ServiceNow ticket", can we do that?

**Comment by debsmita1** [comment]
> Thanks @debsmita1, i have below observations
> 
> image
> * Should the buttons on the sidebar footer be sticky?
> * Will sidebar content change depending on GIt or service now selection
> * Seeing warnings on load

Done
Screenshot 2024-06-10 at 5 45 12 PM
cc @ShiranHi 

@invincibleJai Could you check again if you still see the warning ?

**Comment by invincibleJai** [comment]
@debsmita1 I still see the warnings in dev mode and sorting has no effect on column `catalog-info.yaml`


image

File: plugins/bulk-import/src/components/AddRepositories/RepositoriesTable.tsx [hunk 1]
```diff
@@ -137,55 +135,59 @@ export const RepositoriesTable = ({
...
+    [setFieldValue, values],
   );
   const handleClickAllForDrawerSelection = (checked: boolean) => {
     if (updateSelectedReposInDrawer) {
       if (checked) {
         // Select all repos that don't have the status 'Exists', set added repos id to -1
         const newSelected = filteredData.reduce(
           (acc, repo) => {
-            acc.push(repo.catalogInfoYaml?.status !== 'Exists' ? repo.id : -1);
+            if (repo.catalogInfoYaml?.status !== 'Exists') {
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/AddRepositories/RepositoriesTable.tsx:188]: nit: should we move this to const as used multiple times/places

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 2]
```diff
@@ -2,67 +2,22 @@ import React from 'react';
...
   a: AddRepositoriesData,
   b: AddRepositoriesData,
   orderBy: string,
 ) => {
-  const value1 = get(a, orderBy);
-  const value2 = get(b, orderBy);
+  let value1 = get(a, orderBy);
+  let value2 = get(b, orderBy);
+
+  if (orderBy === 'selectedRepositories') {
```
Comment (invincibleJai) [inline plugins/bulk-import/src/utils/repository-utils.tsx:17]: do we need the specific check?

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 3]
```diff
@@ -86,44 +41,63 @@ export const createData = (
...
+        org.repositories.push(repo);
+      } else {
+        acc.push({
+          id,
+          orgName: repo.organizationUrl,
+          organizationUrl: repo.organizationUrl,
+          repositories: [repo],
+          selectedRepositories: [],
+        });
+        id++;
```
Comment (invincibleJai) [inline plugins/bulk-import/src/utils/repository-utils.tsx:84]: can we use some unique id here from data i.e `repo.id` or if you need index then reducer does return index as well.

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 4]
```diff
@@ -174,5 +168,39 @@ export const filterSelectedForActiveDrawer = (
 };
 
 export const urlHelper = (url: string) => {
+  if (!url || url === '') {
+    return url;
```
Comment (invincibleJai) [inline plugins/bulk-import/src/utils/repository-utils.tsx:172]: if url is not there or empty we are returning the same value, should we return null/undefined or some value?

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 5]
```diff
@@ -174,5 +168,39 @@ export const filterSelectedForActiveDrawer = (
...
+  }
   return url.split('https://')[1] || url;
 };
+
+export const getNewOrgsData = (
```
Comment (invincibleJai) [inline plugins/bulk-import/src/utils/repository-utils.tsx:177]: good to add unit test for utils

**Review by invincibleJai — CHANGES_REQUESTED** [review]
Thanks @debsmita1 , overall looks good. Added some comments see if it makes sense

+1 for e2e and will good to add unit tests as well

File: plugins/bulk-import/src/utils/repository-utils.tsx (hunk 2 above)
Comment (debsmita1) [reply plugins/bulk-import/src/utils/repository-utils.tsx:17]: The `selectedRepositories` value is an array, I am sorting on the basis of the array length

File: plugins/bulk-import/src/utils/repository-utils.tsx (hunk 4 above)
Comment (debsmita1) [reply plugins/bulk-import/src/utils/repository-utils.tsx:172]: returning '-'

File: plugins/bulk-import/src/utils/repository-utils.tsx (hunk 5 above)
Comment (debsmita1) [reply plugins/bulk-import/src/utils/repository-utils.tsx:177]: Done

File: plugins/bulk-import/src/utils/repository-utils.tsx (hunk 3 above)
Comment (debsmita1) [reply plugins/bulk-import/src/utils/repository-utils.tsx:84]: yes, using `repo.id` now

**Comment by debsmita1** [comment]
> @debsmita1 I still see the warnings in dev mode and sorting has no effect on column `catalog-info.yaml`.
> 
> image

This warning can be seen on other plugins as well, would need to dig deeper
Screenshot 2024-06-19 at 9 30 33 PM




> Nit: Texts are overlapping on the sticky footer for legends
> 
> image

Fixed it 

Screenshot 2024-06-19 at 9 32 44 PM

File: plugins/bulk-import/src/components/AddRepositories/AddRepositoriesPage.tsx [hunk 6]
```diff
@@ -2,118 +2,103 @@ import React from 'react';
...
+              <Typography variant="h5">
+                Add repositories to Red Hat Developer Hub in 5 steps
+              </Typography>
+            </AccordionSummary>
+            <AccordionDetails
+              sx={{
+                flexDirection: 'row',
+                display: 'flex',
+                justifyContent: 'space-around',
+                overflow: 'scroll',
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/AddRepositories/AddRepositoriesPage.tsx:48]:
```suggestion
                overflow: 'auto',
```

i can see scroll, always

image

**Review by invincibleJai — APPROVED** [review]

**Pull request merged**